tests/test_mlip_calculators.py

```diff
@@ -244,3 +244,24 @@ def test_invalid_device(arch):
     """Test error raised if invalid device is specified."""
     with pytest.raises(ValueError):
         choose_calculator(arch=arch, device="invalid")
+
+
+def test_d3():
+    """Test adding D3 dispersion calculator automatically."""
+    skip_extras("mace_mp")
+
+    calculator = choose_calculator(arch="mace_mp", dispersion=True)
+    assert calculator.parameters["version"] is not None
+    assert calculator.parameters["model"] is not None
+    assert calculator.parameters["arch"] == "mace_mp_d3"
+
+
+def test_d3_manual():
+    """Test adding D3 dispersion calculator manually."""
+    skip_extras("mace_mp")
+
+    calculator = choose_calculator(arch="mace_mp")
+    calculator = add_dispersion(calculator)
```

Comment:
What happens if you add a dispersion whose `device` is different to that of the calculator?

Reply:
I think that should just work, since it basically just runs each calculation independently, and the calculator is responsible preparing the data for its device.

I can't test easily on Mac since the mace models are tricky to run with mps, but e.g.

```
calculator = choose_calculator(arch="mace_mp", device="cuda")
calculator = add_dispersion(calculator, device="cpu")
```

seemed to be usable in a Colab notebook.